homework-submission/todo-app/src/todos/element.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { faCheck, faTrash, faPencilAlt, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { fab } from '@fortawesome/free-brands-svg-icons'
+import {inject, observer} from 'mobx-react';
+
+
+
+ 
+library.add(fab, faCheck, faTrash, faPencilAlt, faTimes);
+
+@inject('TodosStore')
+@observer
+class Element extends Component {
+    state = {
+        id: this.props.id,
+        description: this.props.description,
+        deadline: this.props.deadline,
+        done: this.props.done,
+        editMode: false,
+        temporarySave: this.props.description
+     }
+
+     throwLineHandle = () => {
+        (!this.state.done) ? this.setState({done: true}) : this.setState({done: false});
+     }
+
+     turnUpdateMode = () => {
+        const temp = this.state.description;
+        this.setState({
+            editMode: !this.state.editMode,
+            temporarySave: temp
+        });
+     }
+
+     onChangeHandle = e => {
+         this.setState({[e.target.name]: e.target.value});
+     }
+
+     submitUpdate = e => {
+        e.preventDefault();
+        this.setState({editMode: !this.state.editMode});
+     }
+
+     handleCancel = () => {
+        this.setState({
+            description: this.state.temporarySave,
+            editMode: !this.state.editMode
+        });
+     }
+
+    render() { 
+        const {id} = this.state;
+        
+        return ( 
+            <li className="list-group-item m-1 p-1 text-md-left font-weight-bold" id={"list_item" + id}>
+                {!this.state.editMode ? this.defaultItemsDisplay() : this.updateItemsDisplay()}
+            </li>
+         );
+    }
+
+    defaultItemsDisplay() {
+        const {id, description, deadline} = this.state;
+        const {TodosStore} = this.props;
+        const item = description + ', ' + deadline;
+        const editAndRemoveButtons = <div className="float-right p-0">
+                                        <span className="btn text-primary float-left p-1 m-1 popup-container" onClick={this.turnUpdateMode}>
+                                            <FontAwesomeIcon icon="pencil-alt" />
+                                            <span className="popup-message">Click to Modify</span>
+                                        </span>
+                                        <span className="btn text-danger float-left p-1 m-1 popup-container" onClick={() => TodosStore.deleteItem(id)}>
+                                            <FontAwesomeIcon icon="trash" />
+                                            <span className="popup-message">Click to Remove</span>
+                                        </span>
+                                     </div>;
+        return this.state.done 
```

Comment:
Can this returned value be a bit shorter by making the condition a bit more specified to the `del` element?